andela/ah-fortem-frontend
# 166980307-View & Edit Profiles

#### Title

View & Update User Profiles.

#### Description
It is important for users to be able to view their profiles and also update them. This is vital as it helps users keep content such as bio, image, first name and last name fresh / always up to date.


#### Type of change








#### How Can This Been Tested?
1. Checkout the to `story/166980307-view-and-edit-profiles` branch
2. Create a `.env` in the root directory and populate it with secret keys provided by [imgur](https://api.imgur.com/oauth2/addclient) in the format provided in the `.env.sample`
3. Run `npm install` && `npm run dev`
4. Login to the app with your credentials and visit the profile page. You will see youe details

![image](https://user-images.githubusercontent.com/12128153/61009745-6c508800-a37c-11e9-8b97-71abd139df0f.png)

5. Click on the `edit profile` button and update any of the fields present with your details

![image](https://user-images.githubusercontent.com/12128153/61009851-cc472e80-a37c-11e9-83a0-586289fea201.png)

6. Hit save changes and wait for the confirmation snackbar that your changes have been saved successfully. then reload the page to confirm the same


#### More context
1. You can also view but not edit someone elses profile. All you need is their username, with that go in check
go to `/profile/<username>` and you will be able to see their profile, but the edit button will not be present for you to update (unless the profile is your's)
![image](https://user-images.githubusercontent.com/12128153/61009870-db2de100-a37c-11e9-83ae-d33f9701d5cb.png)

#### Checklist:

- [x]     - setup profiles reducer
- [x]     - create unit tests for the reducer
- [x]     - setup basic Profiledata component and tests
- [x]     - setup dynamic profile page for viewing dynamic user details
- [x]     - create the modal component & tests
- [x]     - create closeicon for modal component
- [x]     - setup edit profile form
- [x]     - test profile edit form
- [x]     - test render prop component that renders edit form
- [x]     - create and test action creators for updating profile
- [x]     - intergrate action creators with edit component
- [x]     - setup env and webpack to read from the .env file
- [x]     - add tests for the imageupload functionality
- [x]     - setup functionality for handling the redirect if a user is not logged in and wants to view the profile page
- [x]     - setup a loading indicator for loading a profile
- [x]     - write api call for updating a new profile


#### PT stories
[#166980307](https://www.pivotaltracker.com/n/projects/2359479/stories/166980307)
